server/src-lib/Hasura/GraphQL/Schema/Mutation.hs

```diff
@@ -0,0 +1,526 @@
+  -> SelPermInfo          -- ^ select permissions of the table
+  -> m (Maybe (FieldParser n (RQL.AnnUpdG UnpreparedValue)))
+updateTableByPk table fieldName description updatePerms selectPerms = runMaybeT $ do
+  tableName <- qualifiedObjectToName table
+  columns   <- lift   $ tableSelectColumns table selectPerms
+  pkArgs    <- MaybeT $ primaryKeysArguments table selectPerms
+  opArgs    <- MaybeT $ updateOperators table updatePerms
+  selection <- lift $ tableSelectionSet table selectPerms
+  let pkFieldName  = $$(G.litName "pk_columns")
+      pkObjectName = tableName <> $$(G.litName "_pk_columns_input")
+      pkObjectDesc = G.Description $ "primary key columns input for table: " <> G.unName tableName
+      argsParser   = do
+        operators <- opArgs
+        primaryKeys <- P.field pkFieldName Nothing $ P.object pkObjectName (Just pkObjectDesc) pkArgs
+        pure (operators, primaryKeys)
+  pure $ P.subselection fieldName description argsParser selection
+    <&> mkUpdateObject table columns updatePerms . fmap RQL.MOutSinglerowObject
+
+mkUpdateObject
+  :: QualifiedTable
+  -> [PGColumnInfo]
+  -> UpdPermInfo
+  -> ( ( [(PGCol, RQL.UpdOpExpG UnpreparedValue)]
+       , AnnBoolExp UnpreparedValue
+       )
+     , RQL.MutationOutputG UnpreparedValue
+     )
+  -> RQL.AnnUpdG UnpreparedValue
+mkUpdateObject table columns updatePerms ((opExps, whereExp), mutationOutput) =
+  RQL.AnnUpd { RQL.uqp1Table   = table
+             , RQL.uqp1OpExps  = opExps
+             , RQL.uqp1Where   = (permissionFilter, whereExp)
+             , RQL.uqp1Check   = checkExp
+             , RQL.uqp1Output  = mutationOutput
+             , RQL.uqp1AllCols = columns
+             }
+  where
+    permissionFilter = fmapAnnBoolExp partialSQLExpToUnpreparedValue $ upiFilter updatePerms
+    checkExp = maybe annBoolExpTrue (fmapAnnBoolExp partialSQLExpToUnpreparedValue) $ upiCheck updatePerms
+
+-- | Various update operators
+updateOperators
+  :: forall m n r. (MonadSchema n m, MonadTableInfo r m)
+  => QualifiedTable -- ^ qualified name of the table
+  -> UpdPermInfo    -- ^ update permissions of the table
+  -> m (Maybe (InputFieldsParser n [(PGCol, RQL.UpdOpExpG UnpreparedValue)]))
+updateOperators table updatePermissions = do
+  tableName <- qualifiedObjectToName table
+  columns   <- tableUpdateColumns table updatePermissions
+  let numericCols = onlyNumCols   columns
+      jsonCols    = onlyJSONBCols columns
+  parsers <- catMaybes <$> sequenceA
+    [ updateOperator tableName $$(G.litName "_set")
+        columnParser RQL.UpdSet columns
+        "sets the columns of the filtered rows to the given values"
+        (G.Description $ "input type for updating data in table " <>> table)
+
+    , updateOperator tableName $$(G.litName "_inc")
+        columnParser RQL.UpdInc numericCols
+        "increments the integer columns with given value of the filtered values"
+        (G.Description $"input type for incrementing integer columns in table " <>> table)
+
```

Comment:
The description and code are not matching here, while the description says it's for incrementing integer columns 
the code is making a `columnParser` out of `numericCols` which includes all the numeric types. 

I tried using the `_inc` operator on a double type, it worked fine. I guess, only the description needs to be changed?

Reply:
That's the same description as on master, though... Is this another case where we improved the code compared to master? @abooij you worked on this IIRC?

Reply:
I don't know the background of where these descriptions came from, or what they should be.

Reply:
I'll just change the description, then; I think that's a harmless enough change.

Reply:
Done, resolving.